### PR TITLE
Check ISymbolRenamedCodeActionOperationFactoryWorkspaceService is null when rename 

### PR DESCRIPTION
**Pull request by Cosifne**
Port https://dev.azure.com/dnceng/internal/_git/dotnet-roslyn/pullrequest/30906

### CODE REVIEW

File: src/Analyzers/Core/CodeFixes/NamingStyle/NamingStyleCodeFixProvider.cs [hunk 1]
```diff
@@ -163,12 +163,17 @@ protected override async Task<IEnumerable<CodeActionOperation>> ComputeOperation
 #if CODE_STYLE  // https://github.com/dotnet/roslyn/issues/42218 tracks removing this conditional code.
                 return SpecializedCollections.SingletonEnumerable(codeAction);
 #else
-                var factory = _startingSolution.Services.GetRequiredService<ISymbolRenamedCodeActionOperationFactoryWorkspaceService>();
-                return new CodeActionOperation[]
+
+                using var _ = PooledObjects.ArrayBuilder<CodeActionOperation>.GetInstance(out var operations);
```
Comment (CyrusNajmabadi) [inline src/Analyzers/Core/CodeFixes/NamingStyle/NamingStyleCodeFixProvider.cs:167]:
```suggestion
                using var operations = TemporaryArray<CodeActionOperation>.Empty;
```

File: src/Analyzers/Core/CodeFixes/NamingStyle/NamingStyleCodeFixProvider.cs [hunk 2]
```diff
@@ -163,12 +163,17 @@ protected override async Task<IEnumerable<CodeActionOperation>> ComputeOperation
...
+                var factory = _startingSolution.Services.GetService<ISymbolRenamedCodeActionOperationFactoryWorkspaceService>();
+                if (factory is not null)
                 {
-                    codeAction,
-                    factory.CreateSymbolRenamedOperation(_symbol, _newName, _startingSolution, newSolution)
-                };
+                    operations.Add(factory.CreateSymbolRenamedOperation(_symbol, _newName, _startingSolution, newSolution));
+                }
+
+                return operations.ToImmutable();
```
Comment (CyrusNajmabadi) [inline src/Analyzers/Core/CodeFixes/NamingStyle/NamingStyleCodeFixProvider.cs:176]:
```suggestion
                return operations.ToImmutableAndClear();
```

File: src/Analyzers/Core/CodeFixes/NamingStyle/NamingStyleCodeFixProvider.cs (hunk 1 above)
Comment (ryzngard) [reply src/Analyzers/Core/CodeFixes/NamingStyle/NamingStyleCodeFixProvider.cs:167]: ❓ what's the difference between the two?

**Review by ryzngard — APPROVED** [review]

File: src/Analyzers/Core/CodeFixes/NamingStyle/NamingStyleCodeFixProvider.cs (hunk 1 above)
Comment (Cosifne) [reply src/Analyzers/Core/CodeFixes/NamingStyle/NamingStyleCodeFixProvider.cs]: I believe it is design to held small amount of objects.

Comment (CyrusNajmabadi) [reply src/Analyzers/Core/CodeFixes/NamingStyle/NamingStyleCodeFixProvider.cs]: Yup. For when you know you have 4 or fewer elements, it's more efficient.

**Pull request merged**